### ISSUE TITLE
some questions about fine-tuning and matching loss

### ISSUE THREAD

**Issue by EliotChenKJ** [grey-eye/talking-heads#40]
Hello~I have two questions about matching loss:

- In my opinion, the matrix W and the pose-ignorance code are acutually used in two ways. W is used as the projected embeddings of classes in projection cGAN, while the other serves as the "identity code". However, i don't actually understand **what it means that we choose to make Wi closer to the identity code,** and **how it'll affect the fine-tuning process**.

- the dataset used in fine-tuning: if we've meta-trained the network with large dataset and we would like to fine-tune it with one personal image (One-shot), does that means that we only have one training data in fine-tuning phase?  Then how can we train the network in cGAN way? Consisitently feeding this data into D as real data?

I will appreciate that if you can share your opinions with me~

**Comment by EliotChenKJ** [comment]
Oops, one more Q:
- Why do we not use match loss when fine-tuning?

**Comment by castelo-software** [comment]
Hi @EliotChenKJ, I'll try to answer your questions, but these are just my guesses or how I intepreted these things, I am not an expert:

The Discriminator does a similar encoding as the Embedder, using a given picture and some facial landmarks, and then compares them with a vector (contained in W) in order to say whether the picture and landmarks match. If there's no matching loss, the vectors in W will not necessarily be similar to the results of the Embedder. When fine tuning, the authors scrap W, and instead use the same vector produced by the Embedder to check that the Discriminator's encoding correspond to a matching pair. The way I understood their reasoning, is that if D is trained to work with W vectors that in no way resemble E vectors, D won't give proper results when executed with E vectors in the fine-tuning phase. I hope that's clear enough.

In the fine-tuning process you train with only one data point if you only have one frame, yes. You just do a lot of epochs with the same sample. You're training the network to work only with that one image, so that's how it works. They do say it doesn't work as well with one shot as it does with a few more, so keep that in mind.

And there's no W matrix when fine tuning. You replace it with the vectors that E spits out.

**Comment by EliotChenKJ** [comment]
Thanks for your reply, it is so nice of u to answer these questions patiently~Now i understand it.
